engine/src/main/java/org/terasology/rendering/primitives/ChunkMesh.java

```diff
@@ -275,9 +275,7 @@ private void render(RenderPhase type) {
                 break;
             case ALPHA_REJECT:
                 renderVbo(1);
-                glDisable(GL_CULL_FACE);
                 renderVbo(2);
-                glEnable(GL_CULL_FACE);
```

Comment:
Ok, first of all, I can see how this is causing the problem you mentioned @readv. This is the kind of thing that should be done only via StateChange objects rather than direct glCalls. And if glCalls are really needed they should leave the opengl state as they found it.

Now, from the code it seems that renderVbo(1) (translucent blocks I think) is meant to have face culling ENabled. So, _theoretically_, rather than removing those GL_CULL_FACE lines, you could move the glEnable line over the renderVbo(1) line. 

In practice, I think translucent blocks are for things such as leaves. And if that's the case we should see the back face of a volume of leaves through the front face. So, even in that case I feel face culling should be disabled. 

In short, I think you did the right thing. :smile: 

I'll do some testing tomorrow morning and if all is well we should be able to merge it.

Reply:
That was my line of thinking as well. I had a suspicition that leaves would be rendered incorrectly with the old code, but I couldn't find any translucent leaves block. Today I downloaded JoshariasSurvival module and indeed, the berry bushes were rendered incorrectly. With the new code, they are rendered correctly. Glad to see this helped fix more than just the wireframe mode :)